city_scrapers/spiders/il_finance_authority.py

```diff
@@ -0,0 +1,171 @@
+import re
+from datetime import datetime
+from io import BytesIO, StringIO
+
+import scrapy
+from city_scrapers_core.constants import BOARD, COMMISSION, COMMITTEE
+from city_scrapers_core.items import Meeting
+from city_scrapers_core.spiders import CityScrapersSpider
+from pdfminer.high_level import extract_text_to_fp
+from pdfminer.layout import LAParams
+from pdfminer.pdfparser import PDFSyntaxError
+
+
+class IlFinanceAuthoritySpider(CityScrapersSpider):
+    name = "il_finance_authority"
+    agency = "Illinois Finance Authority"
+    timezone = "America/Chicago"
+    start_urls = ["https://www.il-fa.com/public-access/board-documents/"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def parse(self, response):
+        for item in response.css("tr:nth-child(1n+2)"):
+            pdf_link = self._get_pdf_link(item)
+            if pdf_link is None or not pdf_link.endswith(".pdf"):
+                continue
+            title = self._parse_title(item)
+            date = self._parse_date(item)
+
+            yield scrapy.Request(
+                response.urljoin(pdf_link),
+                callback=self._parse_schedule,
+                dont_filter=True,
+                meta={"title": title, "date": date},
+            )
+
+    def _parse_schedule(self, response):
+        """Parse PDF and then yield to meeting items"""
+        pdf_text = self._parse_agenda_pdf(response)
+        location = self._parse_location(pdf_text)
+        time = self._parse_start(pdf_text)
+        meeting_dict = dict()
+        meeting_dict["title"] = response.meta["title"]
+        meeting_dict["date"] = response.meta["date"]
+        meeting_dict["location"] = location
+        meeting_dict["time"] = time
+
+        yield scrapy.Request(
```

Comment:
Is this last request necessary? It looks like we're trying to `yield` the meeting after parsing the PDF, but is there other information here that we need to get from `_parse_meeting`?

Reply:
The last request is necessary for the code to work.

Reply:
Could you explain more about what you mean? When you create a new request to `response.url`, it looks like you're submitting a second request to the URL you parsed the response from rather than going to a new page